app/Http/Controllers/AuthController.php

```diff
@@ -4,17 +4,67 @@
 
 use App\Models\Config;
 use App\Models\Store;
+use App\Models\User;
+use App\Services\BigCommerceService;
 use GuzzleHttp\Exception\RequestException;
 use Illuminate\Http\Request;
 use Illuminate\Http\Response;
 
 class AuthController extends Controller
 {
-    public function index()
+    /**
+     * App load url as defined in app configuration
+     */
+    public function load(Request $request, BigCommerceService $bigCommerceService)
     {
-        return new Response('github/plugin-bigcommerce-di');
+        $this->validate($request, ['signed_payload' => 'required']);
+        $signedPayload = $request->input('signed_payload');
+
+        $data = $bigCommerceService->verifySignedRequest($signedPayload);
+        $store = Store::where('context', $data['context'])->first();
```

Comment:
There is [`findOrFail`](https://laravel.com/docs/8.x/eloquent#not-found-exceptions) :)